Sources/StreamChat/Repositories/ConnectionRepository.swift

```diff
@@ -141,8 +141,7 @@ class ConnectionRepository {
             shouldNotifyConnectionIdWaiters = true
             connectionId = id
 
-        case let .disconnecting(source) where source.serverError?.isInvalidTokenError == true,
-             let .disconnected(source) where source.serverError?.isInvalidTokenError == true:
+        case let .disconnected(source) where source.serverError?.isInvalidTokenError == true:
```

Comment:
why are we removing the `disconnecting` case? Too many token provider calls?

Reply:
Yes, we are only interested when the websocket finishes disconnecting. The way it was before it would trigger two refresh calls always, which does not make sense